This causes shells in GNOME-Terminal to source /etc/profile and as such have
the right umask (0022 instead of 0007).

diff --git a/src/org.gnome.Terminal.gschema.xml b/src/org.gnome.Terminal.gschema.xml
index e4e0c00..88b4126 100644
--- src/org.gnome.Terminal.gschema.xml
+++ src/org.gnome.Terminal.gschema.xml
@@ -243,7 +243,7 @@
       <description>Possible values are “close” to close the terminal, “restart” to restart the command, and “hold” to keep the terminal open with no command running inside.</description>
     </key>
     <key name="login-shell" type="b">
-      <default>false</default>
+      <default>true</default>
       <summary>Whether to launch the command in the terminal as a login shell</summary>
       <description>If true, the command inside the terminal will be launched as a login shell (argv[0] will have a hyphen in front of it).</description>
     </key>
